sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2ColumnsResourceImpl.java

```diff
@@ -0,0 +1,282 @@
+    }
+    return callWithTable(
+        blockingStub,
+        keyspaceName,
+        tableName,
+        (tableDef) -> {
+          List<Sgv2ColumnDefinition> columns = table2columns(tableDef);
+          final Object payload = raw ? columns : new Sgv2RESTResponse(columns);
+          return jaxrsResponse(Response.Status.OK).entity(payload).build();
+        });
+  }
+
+  @Override
+  public Response createColumn(
+      StargateGrpc.StargateBlockingStub blockingStub,
+      String keyspaceName,
+      String tableName,
+      Sgv2ColumnDefinition columnDefinition,
+      HttpServletRequest request) {
+    if (isStringEmpty(keyspaceName)) {
+      return jaxrsBadRequestError("keyspaceName must be provided").build();
+    }
+    if (isStringEmpty(tableName)) {
+      return jaxrsBadRequestError("table name must be provided").build();
+    }
+    final String columnName = columnDefinition.getName();
+    if (isStringEmpty(tableName)) {
+      return jaxrsBadRequestError("columnName must be provided").build();
+    }
+    return callWithTable(
+        blockingStub,
+        keyspaceName,
+        tableName,
+        (tableDef) -> {
+          Column.Kind kind =
+              columnDefinition.getIsStatic() ? Column.Kind.STATIC : Column.Kind.REGULAR;
+          Column columnDef =
+              ImmutableColumn.builder()
+                  .name(columnName)
+                  .kind(kind)
+                  .type(columnDefinition.getTypeDefinition())
+                  .build();
+          String cql =
+              new QueryBuilder()
+                  .alter()
+                  .table(keyspaceName, tableName)
+                  .addColumn(columnDef)
+                  .build();
+          blockingStub.executeQuery(
+              QueryOuterClass.Query.newBuilder()
+                  .setParameters(parametersForLocalQuorum())
+                  .setCql(cql)
+                  .build());
+
+          return jaxrsResponse(Response.Status.CREATED)
+              .entity(Collections.singletonMap("name", columnName))
+              .build();
+        });
+  }
+
+  @Override
+  public Response getOneColumn(
+      StargateGrpc.StargateBlockingStub blockingStub,
+      String keyspaceName,
+      String tableName,
+      String columnName,
+      boolean raw,
+      HttpServletRequest request) {
+    if (isStringEmpty(keyspaceName)) {
+      return jaxrsBadRequestError("keyspaceName must be provided").build();
+    }
+    if (isStringEmpty(tableName)) {
+      return jaxrsBadRequestError("table name must be provided").build();
+    }
+    if (isStringEmpty(columnName)) {
+      return jaxrsBadRequestError("columnName must be provided").build();
+    }
+    return callWithTable(
+        blockingStub,
+        keyspaceName,
+        tableName,
+        (tableDef) -> {
+          Sgv2ColumnDefinition column = findColumn(tableDef, columnName);
+          if (column == null) {
+            return jaxrsBadRequestError(
```

Comment:
was this already 400? Always kind of confused in SG if such things should be `404`

Reply:
Yes, I am following what Integration tests check. I agree it's not necessarily consistent or maybe even well-documented. I think that 404 is to be returned when directly trying to access non-existing thing (getKeyspace("nosuchks")), whereas 400 (bad request) would be when it's secondary target (update table where keyspace name is invalid). But ... yeah that's not a great explanation.